cartography/intel/aws/kms.py

```diff
@@ -0,0 +1,331 @@
+    #   ]
+    # }
+    if policy is not None:
+        # get just the policy element and convert to JSON because boto3 returns this as string
+        policy = Policy(json.loads(policy['Policy']))
+        if policy.is_internet_accessible():
+            return {
+                "kms_key": key,
+                "internet_accessible": True,
+                "accessible_actions": list(policy.internet_accessible_actions()),
+            }
+        else:
+            return None
+    else:
+        return None
+
+
+@timeit
+def load_kms_keys(neo4j_session, data, region, current_aws_account_id, aws_update_tag):
+    ingest_keys = """
+    MERGE (kmskey:KMSKey{id:{KeyId}})
+    ON CREATE SET kmskey.firstseen = timestamp(),
+    kmskey.arn = {Arn}, kmskey.creationdate = {CreationDate}
+    SET kmskey.deletiondate = {DeletionDate},
+    kmskey.validto = {ValidTo},
+    kmskey.enabled = {Enabled},
+    kmskey.keystate = {KeyState},
+    kmskey.customkeystoreid = {CustomKeyStoreId},
+    kmskey.cloudhsmclusterid = {CloudHsmClusterId},
+    kmskey.lastupdated = {aws_update_tag},
+    kmskey.region = {Region}
+    WITH kmskey
+    MATCH (aa:AWSAccount{id: {AWS_ACCOUNT_ID}})
+    MERGE (aa)-[r:RESOURCE]->(kmskey)
+    ON CREATE SET r.firstseen = timestamp()
+    SET r.lastupdated = {aws_update_tag}
+    """
+
+    # The owner data returned by the API maps to the aws account nickname and not the IAM user
+    # there doesn't seem to be a way to retreive the mapping but we can get the current context account
+    # so we map to that directly
+
+    for kms_key in data:
+        neo4j_session.run(
+            ingest_keys,
+            KeyId=kms_key["KeyId"],
+            Arn=kms_key["Arn"],
+            CreationDate=str(kms_key["CreationDate"]),
+            DeletionDate=str(kms_key.get("DeletionDate")),
+            ValidTo=str(kms_key.get("ValidTo")),
+            Enabled=kms_key["Enabled"],
+            KeyState=kms_key["KeyState"],
+            CustomKeyStoreId=kms_key.get("CustomKeyStoreId"),
+            CloudHsmClusterId=kms_key.get("CloudHsmClusterId"),
+            Region=region,
+            AWS_ACCOUNT_ID=current_aws_account_id,
+            aws_update_tag=aws_update_tag,
+        )
+
+
+@timeit
+def cleanup_kms(neo4j_session, common_job_parameters):
+    run_cleanup_job('aws_import_kms_cleanup.json', neo4j_session, common_job_parameters)
+
+
+@timeit
+def sync_kms_keys(neo4j_session, boto3_session, region, current_aws_account_id, aws_update_tag):
+    kms_keys = get_kms_key_list(boto3_session, region)
+
+    load_kms_keys(neo4j_session, kms_keys, region, current_aws_account_id, aws_update_tag)
+
+    policy_alias_grants_data = get_kms_key_details(boto3_session, kms_keys, region)
+    load_kms_key_details(neo4j_session, policy_alias_grants_data, region, current_aws_account_id, aws_update_tag)
+
+
+@timeit
+def sync(neo4j_session, boto3_session, regions, current_aws_account_id, aws_update_tag, common_job_parameters):
+    logger.info("Inside KMS")
```

Comment:
This could become a bit verbose. It should be fine to remove

Reply:
@marco-lancini removed verbose log statement in latest commit